client-sdk/ts-web/core/cypress/outputConsoleLogs.ts

```diff
@@ -0,0 +1,42 @@
+/** Usage: add to cypress config */
+export const setupNodeEvents: Cypress.Config['setupNodeEvents'] = (on, config) => {
+    on('task', {
+        consoleLog(stringifiedArray) {
+            console.log('console.log', ...JSON.parse(stringifiedArray))
+            return null
+        },
+        consoleWarn(stringifiedArray) {
+            console.warn('console.warn', ...JSON.parse(stringifiedArray))
+            return null
+        },
+        consoleError(stringifiedArray) {
+            console.error('console.error', ...JSON.parse(stringifiedArray))
+            return null
+        },
+    });
+};
+
+/** Usage: call before a test */
+export function beforeWindowLoadListener() {
+    before(() => {
+        Cypress.on('window:before:load', (w) => {
+            cy.stub(w.console, 'log').callsFake((...args) => consoleTask('consoleLog', ...args));
+            cy.stub(w.console, 'warn').callsFake((...args) => consoleTask('consoleWarn', ...args));
+            cy.stub(w.console, 'error').callsFake((...args) => consoleTask('consoleError', ...args));
+        });
+    });
+}
+
+/**
+ * Workaround: `cy.task('consoleLog', stringifiedArray)` throws because cypress command is inside cypress command.
```

Comment:
what is a cypress command? like cy.task? what would have been the outer one?

Reply:
Most cypress things are a cypress command: cy.visit, cy.task, Cypress.on, cy.get().click, cy.screenshot, .. I don't know if they even have a full list in docs.

The outer one is `Cypress.on('window:before:load', (w) => {`
(in https://github.com/oasisprotocol/oasis-sdk/pull/1246 it was `cy.visit('http://localhost:8080/', { onBeforeLoad(w) {`)

Reply:
This workaround is probably even worth extracting into a separate opensource package.

The main "cypress console log to terminal" package that exists only works in chrome

Reply:
oh, this seems to support all browsers https://www.npmjs.com/package/cypress-terminal-report

https://www.npmjs.com/package/cypress-log-to-output didn't